esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2013,15 +2013,12 @@ void CFileSprayEx::readAndCheckSpraySourceReq(IEspContext& context, MemoryBuffer
                 }
             }
 
-            if (dropZone)
-            {
-                //Based on the tests, the dfuserver only supports the wildcard inside the file name, like '/path/f*'.
-                //The dfuserver throws an error if the wildcard is inside the path, like /p*ath/file.
-                SecAccessFlags permission = getDZFileScopePermissions(context, sourcePlaneReq, path, nullptr);
-                if (permission < SecAccess_Read)
-                    throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Access DropZone Scope %s %s not allowed for user %s (permission:%s). Read Access Required.",
-                        sourcePlaneReq.str(), path, context.queryUserId(), getSecAccessFlagName(permission));
-            }
+            //Based on the tests, the dfuserver only supports the wildcard inside the file name, like '/path/f*'.
+            //The dfuserver throws an error if the wildcard is inside the path, like /p*ath/file.
+            SecAccessFlags permission = getDZFileScopePermissions(context, sourcePlaneReq, path, sourceIPReq, SecAccess_Read);
```

Comment:
Removing the check on if (dropZone) line 2016 (and elsewhere throughout these changes), doesn't look right.

It is conflating 2 different issues:
1) The existence of a valid dropzone definition in the environment (or no match if restrictions are off)
2) LDAP checking of that dropzone based on PLANE::<DROPZONE>::... or failing over to legacy in bare-metal if failover is configured.

If the dropzone in 1) doesn't exist (and was allowed to continue because restrictions off), then it should not continue to scope check.
If the dropzone does exist, it should scope check and that scope checking should failover to legacy physical form if failOverToLegacyPhysicalPerms is on.

NB: it is also wrong in dfurun.cpp, it too should not go on to check if an environment dropzone couldn't be found (and was allowed to proceed because restrictions were off).

Reply:
only if dropzone exists. And as noted above, dfurun.cpp should do same.

Reply:
How about the Foreign File in dfurun.cpp? Should I check legacy physical form there if the failOverToLegacyPhysicalPerms is on?

Reply:
yes, that is not dependent on whether there's a dropzone or not.
The comments above are only related to what to do when there's no dropzone found (which implies restrictions off too).